docassemble/ALWeaver/data/static/draggable_table.js

```diff
@@ -0,0 +1,94 @@
+$(document).on('daPageLoad', function(){            
+    // Grab the field for storing adjusted table order
+    var orderList = $('input[type="draggable_tbl_order_list"]');
+    $(orderList).hide(); 
+    
+    // Grab field scr_table_data (with default value json_string)    
+    var json_data = $('input[type="draggable_tbl_json_string"]');    
+    (json_data).hide();  
+    
+    // If table order is adjusted by user, update draggable_table with data from json_data     
+    var draggable_table = $('.draggable-table');  
+    if (typeof orderList != 'undefined'){
+      if (orderList.val() != '') {
+        var json_records = JSON.parse(json_data.val()); 
+        /* Error "Uncaught SyntaxError: JSON.parse: unexpected character at line 1 column 1 of the JSON data" is seen on screens where draggable-table doesn't exist, why?*/
```

Comment:
Unless you put a `input[type="draggable_tbl_json_string"]` on every page of the interview, `json_data.val()` will be `undefined` and you're basically doing `JSON.parse(undefined)`. That will throw an error. If you want to avoid this, add a test at the beginning. Test whether these elements exist on the page and if they don't then return from the function to stop the rest running.

We actually did decide to wrap any js functionality in try/catch in case such errors happened so they wouldn't block other javascript from working if something was wrong. I think this functionality would be one of those situations.

Reply:
Tried `if (orderList.val() != 'undefined') `, still got the same error. We may need to chat in zoom this afternoon or whenever you have time.

Reply:
Thanks for the tip to use the console for live data! I changed orderList to $(orderList) and that fixed it. 

I'm having trouble pushing the change. Will continue on Wed.

Reply:
In case I don't catch you later, here are some clarifications about your initial findings:

> Tried if (orderList.val() != 'undefined') , still got the same error.
`'undefined'` is a string. If my guess is correct, you'd need `undefined`, which is a javascript value. No quotes.

> Entered $('input[type="draggable_tbl_order_list"]') in the console on a broken page and it's defined.
Then it's what I thought it was. That object you see has a length of 0. That means jQuery found absolutely no nodes matching your selector. That's just how jQuery does things.

Reply:
My bad. I got rid of `typeof` but forgot to get rid of the quotes. Many thanks for the clarifications!